marlowe-playground-client/src/Reachability.purs

```diff
@@ -109,8 +117,20 @@ zipperToContractPathAux HeadZip p = p
 zipperToContractPath :: ContractZipper -> ContractPath
 zipperToContractPath zipper = zipperToContractPathAux zipper Nil
 
-checkContractForReachability :: forall m. String -> String -> HalogenM State Action ChildSlots Message m Unit
-checkContractForReachability contract state = H.raise $ WebSocketMessage $ CheckForWarnings (encodeJSON true) contract state
+runAjax ::
+  forall m a.
+  ExceptT AjaxError (HalogenM State Action ChildSlots Message m) a ->
+  HalogenM State Action ChildSlots Message m (WebData a)
+runAjax action = RemoteData.fromEither <$> runExceptT action
+
+checkContractForReachability ::
+  forall m.
+  MonadAff m =>
+  SPSettings_ SPParams_ ->
+  String ->
+  String ->
```

Comment:
Does this call for newtypes? 🤔

Reply:
not quite, instead it calls for encode at the point of sending rather than before hand, much better thanks!